pkg/restapi/webhook/samplewebhookclient.go

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package webhook
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+)
+
+type sampleWebhookClient struct {
+	addr           string
+	handlerPattern string
+}
+
+func (s *sampleWebhookClient) listenAndStopAfterReceivingNotification() error {
+	srv := &http.Server{Addr: s.addr, Handler: http.DefaultServeMux}
+	http.HandleFunc(s.handlerPattern, func(resp http.ResponseWriter, req *http.Request) {
+		logger.Infof("Just received webhook notification")
+
+		err := srv.Shutdown(context.Background())
+		if err != nil {
+			logger.Fatalf("Failed to shutdown sample webhook client server: %s", err)
```

Comment:
Are you sure of using Fatal here? it calls os.Exit(1), be careful with it.

Reply:
Oops, meant to use Errorf. Fixed!